cylc/flow/network/schema.py

```diff
@@ -349,8 +349,6 @@ async def get_nodes_all(root, info, **args):
 
     _, field_ids = process_resolver_info(root, info, args)
 
-    if hasattr(args, 'id'):
-        args['ids'] = [args.get('id')]
```

Comment:
This logic is flawed because a dict will never have an attribute `id` (only a key, perhaps)

Reply:
This line doesn't appear to have been hit in coverage so this may be correct: https://app.codecov.io/gh/cylc/cylc-flow/blob/master/cylc%2Fflow%2Fnetwork%2Fschema.py#L352

But this could just be a coverage hole. Dunno, @dwsutherland any ideas.

Reply:
It is physically impossible to hit - a dictionary like `foo = {'id': 1}` doesn't have `foo.id`, it has `foo['id']`

Reply:
True if its always a dictionary. By the looks of it, the purpose of this code is to handle an object that isn't a dictionary but supports `__setitem__`.

Reply:
`args` is a dict because it's the keyword arguments object from the function definition

Reply:
Can't fault that!